autosubmit/monitor/monitor.py

```diff
@@ -91,52 +102,90 @@ def _display_file(a_file: str):
+            color = _MONITOR_STATUS_TO_COLOR.get(Status.DELAYED,None)
+        elif job.name in child.edge_info.get("UNKNOWN", {}):
+            color = _MONITOR_STATUS_TO_COLOR.get(Status.UNKNOWN,None)
+        elif job.name in child.edge_info.get("SUSPENDED", {}):
+            color = _MONITOR_STATUS_TO_COLOR.get(Status.SUSPENDED,None)
+        elif job.name in child.edge_info.get("SKIPPED", {}):
+            color = _MONITOR_STATUS_TO_COLOR.get(Status.SKIPPED,None)
+        elif job.name in child.edge_info.get("WAITING", {}):
+            color = _MONITOR_STATUS_TO_COLOR.get(Status.WAITING,None)
+        elif job.name in child.edge_info.get("READY", {}):
+            color = _MONITOR_STATUS_TO_COLOR.get(Status.READY,None)
+        elif job.name in child.edge_info.get("SUBMITTED", {}):
+            color = _MONITOR_STATUS_TO_COLOR.get(Status.SUBMITTED,None)
+        else:
+            return None, None
+        if label and label == 0:
+            label = None
+        return color,label
+    else:
+        return None, None
+
+
 class Monitor:
     """Class to handle monitoring of Jobs at HPC."""
-    _table = dict([(Status.UNKNOWN, 'white'), (Status.WAITING, 'gray'), (Status.READY, 'lightblue'), (Status.PREPARED, 'skyblue'),
-         (Status.SUBMITTED, 'cyan'), (Status.HELD, 'salmon'), (Status.QUEUING, 'pink'), (Status.RUNNING, 'green'), (Status.COMPLETED, 'yellow'), (Status.FAILED, 'red'), (Status.DELAYED, 'lightcyan'),
-         (Status.SUSPENDED, 'orange'), (Status.SKIPPED, 'lightyellow')])
 
     def __init__(self):
         self.nodes_plotted = None
 
-    @staticmethod
-    def color_status(status):
-        """
-        Return color associated to given status
-
-        :param status: status
-        :type status: Status
-        :return: color
-        :rtype: str
-        """
-        if status == Status.WAITING:
-            return Monitor._table[Status.WAITING]
-        elif status == Status.READY:
-            return Monitor._table[Status.READY]
-        elif status == Status.PREPARED:
-            return Monitor._table[Status.PREPARED]
-        elif status == Status.SUBMITTED:
-            return Monitor._table[Status.SUBMITTED]
-        elif status == Status.HELD:
-            return Monitor._table[Status.HELD]
-        elif status == Status.QUEUING:
-            return Monitor._table[Status.QUEUING]
-        elif status == Status.RUNNING:
-            return Monitor._table[Status.RUNNING]
-        elif status == Status.COMPLETED:
-            return Monitor._table[Status.COMPLETED]
-        elif status == Status.SKIPPED:
-            return Monitor._table[Status.SKIPPED]
-        elif status == Status.FAILED:
-            return Monitor._table[Status.FAILED]
-        elif status == Status.SUSPENDED:
-            return Monitor._table[Status.SUSPENDED]
-        elif status == Status.DELAYED:
-            return Monitor._table[Status.SUSPENDED]
-        else:
-            return Monitor._table[Status.UNKNOWN]
```

Comment:
These if/else/etc can be replaced by a simpler dictionary lookup, I believe. Done above.